delfin/drivers/dell_emc/vmax/perf_utils.py

```diff
@@ -12,88 +12,44 @@
-    """
-    return {'symmetrixId': str(array),
-            "endDate": end_time,
-            "startDate": start_time,
-            "metrics": metrics,
-            "dataFormat": "Average"}
-
-
-def parse_performance_data(response):
+def parse_performance_data(metrics):
     """Parse metrics response to a map
-    :param response: response from unispshere REST API
+    :param metrics: metrics from unispshere REST API
     :returns: map with key as metric name and value as dictionary
         containing {timestamp: value} for a the timestamps available
     """
     metrics_map = {}
-    for metrics in response["resultList"]["result"]:
-        timestamp = metrics["timestamp"]
-        for key, value in metrics.items():
-            metrics_map[key] = metrics_map.get(key, {})
-            metrics_map[key][timestamp] = value
+    timestamp = metrics["timestamp"]
+    for key, value in metrics.items():
+        metrics_map[key] = metrics_map.get(key, {})
+        metrics_map[key][timestamp] = value
     return metrics_map
 
 
-def map_array_perf_metrics_to_delfin_metrics(metrics_value_map):
-    """map vmax array performance metrics values  to delfin metrics values
-        :param metrics_value_map: metric to values map of vmax metrics
-        :returns: map with key as delfin metric name and value as dictionary
-            containing {timestamp: value} for a the timestamps available
-        """
-    # read and write response_time
-    read_response_values_dict = metrics_value_map.get('ReadResponseTime')
-    write_response_values_dict = metrics_value_map.get('WriteResponseTime')
-    if read_response_values_dict or write_response_values_dict:
-        response_time_values_dict = \
-            Counter(read_response_values_dict) + \
-            Counter(write_response_values_dict)
-    # bandwidth metrics
-    read_bandwidth_values_dict = metrics_value_map.get('HostMBReads')
-    write_bandwidth_values_dict = metrics_value_map.get('HostMBWritten')
-    if read_bandwidth_values_dict or write_bandwidth_values_dict:
-        bandwidth_values_dict = \
-            Counter(read_bandwidth_values_dict) +\
-            Counter(write_bandwidth_values_dict)
-    throughput_values_dict = metrics_value_map.get('HostIOs')
-    read_throughput_values_dict = metrics_value_map.get('HostReads')
-    write_throughput_values_dict = metrics_value_map.get('HostWrites')
-    # map values to delfin metrics spec
-    delfin_metrics = {'responseTime': response_time_values_dict,
-                      'readThroughput': read_bandwidth_values_dict,
-                      'writeThroughput': write_bandwidth_values_dict,
-                      'requests': throughput_values_dict,
-                      'readRequests': read_throughput_values_dict,
-                      'writeRequests': write_throughput_values_dict,
-                      'throughput': bandwidth_values_dict}
-    return delfin_metrics
+def construct_metrics(storage_id, resource_metrics, unit_map, perf_list):
```

Comment:
Resource_type can be taken as an input here, then  Then we dont have have parse through VMAX_METRICS array, also need for VMAX_METRICS array can be avoided..